src/vs/workbench/common/editor/editorGroupModel.ts

```diff
@@ -4,14 +4,15 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { Event, Emitter } from 'vs/base/common/event';
-import { IEditorFactoryRegistry, GroupIdentifier, EditorsOrder, EditorExtensions, IUntypedEditorInput, SideBySideEditor, IEditorMoveEvent, IEditorOpenEvent, EditorCloseContext, IEditorCloseEvent } from 'vs/workbench/common/editor';
+import { IEditorFactoryRegistry, GroupIdentifier, EditorsOrder, EditorExtensions, IUntypedEditorInput, SideBySideEditor, EditorCloseContext, IEditorCloseEvent } from 'vs/workbench/common/editor';
 import { EditorInput } from 'vs/workbench/common/editor/editorInput';
 import { SideBySideEditorInput } from 'vs/workbench/common/editor/sideBySideEditorInput';
 import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
 import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
 import { dispose, Disposable, DisposableStore } from 'vs/base/common/lifecycle';
 import { Registry } from 'vs/platform/registry/common/platform';
 import { coalesce } from 'vs/base/common/arrays';
+import { GroupChangeKind, IGroupChangeEvent } from 'vs/workbench/services/editor/common/editorGroupsService';
```

Comment:
This sounds like a wrong dependency to me, the model should not reach back into the service and I think it should not use the same event as the UI event.

Reply:
It just seemed like a nice shape and I would just be remaking it. You might argue that the UI event is powered almost completely by the model so it makes sense to be in this shape.

Reply:
I dont mind that both events use the same underlying data structures but then I would suggest the model to define them and the group service to pick them up.